mcpserver/tools/util.go

```diff
@@ -16,6 +16,38 @@ import (
 	"github.com/mattermost/mattermost/server/public/model"
 )
 
+// GetDataDirectoryInternal is the internal function that can be overridden in tests
+var GetDataDirectoryInternal = getDataDirectory
```

Comment:
Maybe we can pass the data directory to `fetchFileData` instead of this override?